src/content/actions/test/receive-profile.js

```diff
@@ -0,0 +1,127 @@
+import { assert } from 'chai';
+import sinon from 'sinon';
+import { blankStore } from './fixtures/stores';
+import * as ProfileViewSelectors from '../../reducers/profile-view';
+import { getView } from '../../reducers/app';
+import { receiveProfileFromAddon, retrieveProfileFromWeb } from '../receive-profile';
+
+import preprocessedProfile from '../../../common/test/fixtures/profile-2d-canvas.json';
+import exampleProfile from '../../../../test/example-profile';
+
+describe('actions/receive-profile', function () {
+  describe('receiveProfileFromAddon', function () {
+    it('can take a profile from an addon and save it to state', function () {
+      const store = blankStore();
+
+      const initialProfile = ProfileViewSelectors.getProfile(store.getState());
+      assert.ok(initialProfile, 'A blank profile initially exists');
+      assert.lengthOf(initialProfile.threads, 0, 'The blank profile contains no data');
+      store.dispatch(receiveProfileFromAddon(preprocessedProfile));
+      assert.strictEqual(ProfileViewSelectors.getProfile(store.getState()), preprocessedProfile, 'The passed in profile is saved in state.');
+    });
+  });
+
+  describe('retrieveProfileFromWeb', function () {
+    let sandbox;
+
+    const fetch404Response = { ok: false, status: 404 };
+    const fetch500Response = { ok: false, status: 500 };
+    const fetch200Response = {
+      ok: true, status: 200,
+      json: () => Promise.resolve(exampleProfile),
+    };
+
+    beforeEach(function () {
+      sandbox = sinon.sandbox.create();
+      sandbox.useFakeTimers();
+
+      global.fetch = sinon.stub();
```

Comment:
I tried another approach where I imported `isomorphic-fetch` and then use the [nock](https://github.com/node-nock/nock) library, but then I thought that here `sinon` was a lot easier.